forge-gui/src/main/java/forge/player/HumanPlay.java

```diff
@@ -623,6 +624,13 @@ else if (ability.getXManaCostPaid() != null) { //ensure pre-announced X value re
             emerge = ability.getSacrificedAsEmerge();
         }
         if (!toPay.isPaid()) {
+            // if matrix still null it's effect payment
+            if (matrix == null) {
+                matrix = new ManaConversionMatrix();
+                matrix.restoreColorReplacements();
```

Comment:
you probably mean restore for manapool? the matrix new created should have restored already

Reply:
no, the problem is new object starts with {0} matrix on `colorRestriction` so nothing can be paid instead of default

maybe we should make that part of the constructor?